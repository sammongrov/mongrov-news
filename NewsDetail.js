/**
 * NewsList Screen
 */

import React, { Component } from 'react';
import { WebView, TouchableOpacity, BackHandler, View, ScrollView } from 'react-native';
import { Actions } from 'react-native-router-flux';
import PropTypes from 'prop-types';
import { styles } from 'react-native-theme';
import AppUtils from '@utils';
import FastImage from 'react-native-fast-image';
import { Icon, Screen, NavBar, Text, Avatar, AntDesignIcon } from '@ui/components';
import { Colors } from '@ui/theme_default';

class NewsDetail extends Component {
  constructor(props) {
    super(props);
    this._news = props.newsUrl;
    this.state = {
      newsUrl: this._news,
    };
  }

  componentWillMount() {
    BackHandler.addEventListener('hardwareBackPress', this.handleBackPress);
  }

  componentWillUnmount = () => {
    BackHandler.removeEventListener('hardwareBackPress', this.handleBackPress);
  };

  handleBackPress = () => {
    Actions.pop();
    return true;
  };

  renderNewsContent() {
    const { newsUrl } = this.state;
    const { newsDetails } = this.props;
    if (newsUrl) {
      return <WebView source={{ uri: newsUrl }} style={{}} />;
    }
    return (
      <ScrollView contentContainerStyle={{ flexGrow: 1, flexDirection: 'column', padding: 10 }}>
        <View
          style={{
            flexDirection: 'row',
            alignItems: 'center',
            justifyContent: 'space-between',
            marginBottom: 15,
          }}
        >
          <View style={{ flexDirection: 'row', alignItems: 'center' }}>
            {newsDetails.user.avatarUrl && (
              <Avatar
                avatarUrl={newsDetails.user.avatarUrl ? newsDetails.user.avatarUrl : null}
                avatarName={newsDetails.user.displayName}
                avatarSize={35}
              />
            )}
            <View style={{ flexDirection: 'column', justifyContent: 'center' }}>
              <Text
                numberOfLines={1}
                ellipsizeMode="tail"
                style={[styles.newsUsernameText, { flex: 1, marginRight: 15, paddingLeft: 5 }]}
              >
                {newsDetails.user.displayName
                  ? newsDetails.user.displayName
                  : newsDetails.user.username}
              </Text>
              <Text style={[styles.newsTimeText, { paddingLeft: 5 }]}>
                {AppUtils.formatNewsDate(newsDetails.postedAt)}
              </Text>
            </View>
          </View>
          <View
            style={{
              flexDirection: 'row',
              alignItems: 'center',
              justifyContent: 'flex-end',
              paddingRight: 10,
            }}
          >
            <TouchableOpacity
              style={{
                flexDirection: 'row',
                alignItems: 'center',
                justifyContent: 'center',
                paddingRight: 30,
              }}
            >
              <AntDesignIcon name="like2" color={Colors.ICON_COMMENT} size={16} />
              <Text style={[styles.newsStatisticsText]}>
                {newsDetails.baseScore !== null ? newsDetails.baseScore : 0}
              </Text>
            </TouchableOpacity>
            <TouchableOpacity
              style={{ flexDirection: 'row', alignItems: 'center', justifyContent: 'center' }}
              onPress={() =>
                Actions.NewsComment({
                  newsDetails,
                })
              }
            >
              <AntDesignIcon name="message1" color={Colors.ICON_COMMENT} size={14} />
              <Text style={[styles.newsStatisticsText]}>
                {newsDetails.commentCount !== null ? newsDetails.commentCount : 0}
              </Text>
            </TouchableOpacity>
          </View>
        </View>
        <View style={{ flexDirection: 'column' }}>
          {newsDetails.title && (
            <Text style={[styles.newsTitleText, { fontSize: 22 }]}>{newsDetails.title}</Text>
          )}
          {newsDetails.thumbnailUrl && (
            <View style={styles.newsImageView}>
              <FastImage
                style={styles.newsImageHeight}
                source={{
                  uri: newsDetails.thumbnailUrl ? newsDetails.thumbnailUrl : newsDetails.avatarUrl,
                  priority: FastImage.priority.high,
                }}
                resizeMode={FastImage.resizeMode.contain}
              />
            </View>
          )}
          <View style={[styles.newsExcerptView]}>
            <Text
              style={[
                styles.newsExcerptText,
                newsDetails.thumbnailUrl || !newsDetails.title ? null : styles.marginTop10,
              ]}
            >
              {newsDetails.body}
            </Text>
          </View>
        </View>
        <View
          style={{
            flex: 1,
            flexDirection: 'row',
            alignItems: 'flex-end',
            paddingBottom: 20,
          }}
        >
          {newsDetails.viewCount && (
            <TouchableOpacity
              style={{ flexDirection: 'row', alignItems: 'center', paddingRight: 20 }}
            >
              <AntDesignIcon name="eyeo" color={Colors.ICON_COMMENT} size={23} />
              <Text style={styles.newsInfoText}>
                {newsDetails.viewCount !== null ? newsDetails.viewCount : 0}
              </Text>
            </TouchableOpacity>
          )}
          {newsDetails.clickCount && (
            <TouchableOpacity style={{ flexDirection: 'row', alignItems: 'center' }}>
              <AntDesignIcon name="select1" color={Colors.ICON_COMMENT} size={21} />
              <Text style={styles.newsInfoText}>
                {newsDetails.clickCount !== null ? newsDetails.clickCount : 0}
              </Text>
            </TouchableOpacity>
          )}
        </View>
      </ScrollView>
    );
  }

  render() {
    return (
      <Screen>
        <NavBar
          titleText="Article"
          titleTextLine={1}
          titleContainer={{ alignItems: 'flex-start' }}
          leftComponent={
            <TouchableOpacity
              onPress={Actions.pop}
              style={[styles.navSideButtonDimension, styles.alignJustifyCenter]}
            >
              <Icon
                name="chevron-left"
                type="material-community"
                color={Colors.NAV_ICON}
                size={36}
              />
            </TouchableOpacity>
          }
        />
        {this.renderNewsContent()}
      </Screen>
    );
  }
}

export default NewsDetail;

NewsDetail.defaultProps = {
  newsUrl: '',
  newsTitle: '',
  newsExcerpt: '',
  newsThumbnailUrl: '',
  newsDetails: {},
};

NewsDetail.propTypes = {
  newsUrl: PropTypes.string,
  newsTitle: PropTypes.string,
  newsExcerpt: PropTypes.string,
  newsThumbnailUrl: PropTypes.string,
  newsDetails: PropTypes.object,
};
