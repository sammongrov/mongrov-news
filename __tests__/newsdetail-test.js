import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Actions } from 'react-native-router-flux';
import { BackHandler } from 'react-native';
import NewsDetailView from '../NewsDetail';
import NewsDetail from '../NewsDetail';

configure({ adapter: new Adapter() });

jest.mock('react-native-router-flux', () => ({
  Actions: {
    pop: jest.fn(),
    NewsComment: jest.fn(),
  },
}));

jest.mock('BackHandler', () => {
  const backHandler = {
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
  };
  return backHandler;
});

beforeEach(() => {
  jest.resetModules();
});

let newsDetails = {
  user: {
    avatarUrl: 'test_avatarUrl', // false ''
    displayName: 'test_displayName', // false ''
    username: 'test_userName',
  },
  postedAt: 'test_postedAt',
  baseScore: 1, // null
  commentCount: 2, // null
  title: 'test_title', // false ''
  thumbnailUrl: 'test_thumbnailUrl', // false ''
  body: 'test_body',
  viewCount: 3, // null
  clickCount: 4, // null
};

const newsUrl = 'test_newsUrl';

it('news detail renders correctly', () => {
  const rendered = renderer.create(<NewsDetailView newsDetails={newsDetails} />).toJSON();

  expect(rendered).toBeTruthy();
});

it('news detail snapshot test', () => {
  const rendered = renderer.create(<NewsDetailView newsDetails={newsDetails} />).toJSON();

  expect(rendered).toMatchSnapshot();
});

it('newsDetail unmounts correctly', () => {
  const newsDetailTree = shallow(<NewsDetail newsDetails={newsDetails} />);
  newsDetailTree.unmount();

  expect(BackHandler.removeEventListener).toHaveBeenCalled();
});

it('handleBackPress', () => {
  const rootComponent = shallow(<NewsDetail newsDetails={newsDetails} />);
  const instance = rootComponent.instance();
  const result = instance.handleBackPress();

  expect(result).toBe(true);
  expect(Actions.pop.mock.calls.length).toBe(1);
});

it('First TouchOP Actions is called', () => {
  const rootComp = shallow(<NewsDetail newsDetails={newsDetails} />);
  const navBar = rootComp.find('NavBar').first();
  const TouchOp = navBar
    .shallow()
    .find('TouchableOpacity')
    .first();
  TouchOp.props().onPress();

  expect(Actions.pop).toBeCalled();
});

it('TouchOp Actions is called in renderNewsContent', () => {
  const rootComp = shallow(<NewsDetail newsDetails={newsDetails} />);
  const TO = rootComp.find({ name: 'message1' }).parent();
  TO.props().onPress();

  expect(Actions.NewsComment).toBeCalled();
});

it('renderNewsContent first if', () => {
  const rootComp = shallow(<NewsDetail newsDetails={newsDetails} newsUrl={newsUrl} />);
  const webView = rootComp.find('WebView').first();

  expect(webView).toBeTruthy();
});

it('covered all branches with false/null', () => {
  newsDetails = {
    user: {
      avatarUrl: '', // false
      displayName: '', // false ''
      username: 'test_userName',
    },
    postedAt: 'test_postedAt',
    baseScore: null, // null
    commentCount: null, // null
    title: '', // false ''
    thumbnailUrl: '', // false ''
    body: 'test_body',
    viewCount: null, // null
    clickCount: null, // null
  };
  const newsDetailTree = renderer.create(<NewsDetail newsDetails={newsDetails} />).toJSON();

  expect(newsDetailTree).toMatchSnapshot();
});
