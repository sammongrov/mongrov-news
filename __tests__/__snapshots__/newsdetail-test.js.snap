// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`covered all branches with false/null 1`] = `
<BVLinearGradient
  colors={
    Array [
      4293914607,
      4293914607,
    ]
  }
  locations={
    Array [
      0,
      1,
    ]
  }
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
    >
      <View
        style={
          Array [
            undefined,
            null,
          ]
        }
      >
        <View
          style={
            Array [
              undefined,
              undefined,
            ]
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#2E88FF",
                    "fontSize": 36,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              type="material-community"
            >
              
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              undefined,
              undefined,
              Object {
                "alignItems": "flex-start",
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "fontFamily": "OpenSans-Regular",
                },
                Array [
                  undefined,
                  null,
                ],
              ]
            }
          >
            Article
          </Text>
        </View>
        <View
          style={
            Array [
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <RCTScrollView
        contentContainerStyle={
          Object {
            "flexDirection": "column",
            "flexGrow": 1,
            "padding": 10,
          }
        }
      >
        <View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginBottom": 15,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              
              <View
                style={
                  Object {
                    "flexDirection": "column",
                    "justifyContent": "center",
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  numberOfLines={1}
                  style={
                    Array [
                      Object {
                        "fontFamily": "OpenSans-Regular",
                      },
                      Array [
                        undefined,
                        Object {
                          "flex": 1,
                          "marginRight": 15,
                          "paddingLeft": 5,
                        },
                      ],
                    ]
                  }
                >
                  test_userName
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "fontFamily": "OpenSans-Regular",
                      },
                      Array [
                        undefined,
                        Object {
                          "paddingLeft": 5,
                        },
                      ],
                    ]
                  }
                >
                  Invalid date
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "flex-end",
                  "paddingRight": 10,
                }
              }
            >
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "opacity": 1,
                    "paddingRight": 30,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#4C6070",
                        "fontSize": 16,
                      },
                      undefined,
                      Object {
                        "fontFamily": "anticon",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "fontFamily": "OpenSans-Regular",
                      },
                      Array [
                        undefined,
                      ],
                    ]
                  }
                >
                  0
                </Text>
              </View>
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "opacity": 1,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#4C6070",
                        "fontSize": 14,
                      },
                      undefined,
                      Object {
                        "fontFamily": "anticon",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "fontFamily": "OpenSans-Regular",
                      },
                      Array [
                        undefined,
                      ],
                    ]
                  }
                >
                  0
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "flexDirection": "column",
              }
            }
          >
            
            
            <View
              style={
                Array [
                  undefined,
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "fontFamily": "OpenSans-Regular",
                    },
                    Array [
                      undefined,
                      null,
                    ],
                  ]
                }
              >
                test_body
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "flex": 1,
                "flexDirection": "row",
                "paddingBottom": 20,
              }
            }
          />
        </View>
      </RCTScrollView>
    </View>
  </RCTSafeAreaView>
</BVLinearGradient>
`;

exports[`news detail snapshot test 1`] = `
<BVLinearGradient
  colors={
    Array [
      4293914607,
      4293914607,
    ]
  }
  locations={
    Array [
      0,
      1,
    ]
  }
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
    >
      <View
        style={
          Array [
            undefined,
            null,
          ]
        }
      >
        <View
          style={
            Array [
              undefined,
              undefined,
            ]
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#2E88FF",
                    "fontSize": 36,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              type="material-community"
            >
              
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              undefined,
              undefined,
              Object {
                "alignItems": "flex-start",
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "fontFamily": "OpenSans-Regular",
                },
                Array [
                  undefined,
                  null,
                ],
              ]
            }
          >
            Article
          </Text>
        </View>
        <View
          style={
            Array [
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <RCTScrollView
        contentContainerStyle={
          Object {
            "flexDirection": "column",
            "flexGrow": 1,
            "padding": 10,
          }
        }
      >
        <View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginBottom": 15,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  Array [
                    undefined,
                    Object {
                      "position": "relative",
                    },
                  ]
                }
              >
                <View
                  accessible={true}
                  isTVSelectable={true}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "backgroundColor": "#74c22e",
                      "borderRadius": 17.5,
                      "height": 35,
                      "opacity": 1,
                      "width": 35,
                    }
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "overflow": "hidden",
                        },
                        Object {
                          "borderRadius": 17.5,
                          "height": 35,
                          "width": 35,
                        },
                      ]
                    }
                  >
                    <FastImageView
                      onFastImageError={[Function]}
                      resizeMode="cover"
                      source={
                        Object {
                          "priority": "high",
                          "uri": "test_avatarUrl",
                        }
                      }
                      style={
                        Object {
                          "bottom": 0,
                          "left": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        }
                      }
                    />
                  </View>
                </View>
              </View>
              <View
                style={
                  Object {
                    "flexDirection": "column",
                    "justifyContent": "center",
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  numberOfLines={1}
                  style={
                    Array [
                      Object {
                        "fontFamily": "OpenSans-Regular",
                      },
                      Array [
                        undefined,
                        Object {
                          "flex": 1,
                          "marginRight": 15,
                          "paddingLeft": 5,
                        },
                      ],
                    ]
                  }
                >
                  test_displayName
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "fontFamily": "OpenSans-Regular",
                      },
                      Array [
                        undefined,
                        Object {
                          "paddingLeft": 5,
                        },
                      ],
                    ]
                  }
                >
                  Invalid date
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "flex-end",
                  "paddingRight": 10,
                }
              }
            >
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "opacity": 1,
                    "paddingRight": 30,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#4C6070",
                        "fontSize": 16,
                      },
                      undefined,
                      Object {
                        "fontFamily": "anticon",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "fontFamily": "OpenSans-Regular",
                      },
                      Array [
                        undefined,
                      ],
                    ]
                  }
                >
                  1
                </Text>
              </View>
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "opacity": 1,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#4C6070",
                        "fontSize": 14,
                      },
                      undefined,
                      Object {
                        "fontFamily": "anticon",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "fontFamily": "OpenSans-Regular",
                      },
                      Array [
                        undefined,
                      ],
                    ]
                  }
                >
                  2
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "flexDirection": "column",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "fontFamily": "OpenSans-Regular",
                  },
                  Array [
                    undefined,
                    Object {
                      "fontSize": 22,
                    },
                  ],
                ]
              }
            >
              test_title
            </Text>
            <View>
              <View
                style={
                  Array [
                    Object {
                      "overflow": "hidden",
                    },
                    undefined,
                  ]
                }
              >
                <FastImageView
                  resizeMode="contain"
                  source={
                    Object {
                      "priority": "high",
                      "uri": "test_thumbnailUrl",
                    }
                  }
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                />
              </View>
            </View>
            <View
              style={
                Array [
                  undefined,
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "fontFamily": "OpenSans-Regular",
                    },
                    Array [
                      undefined,
                      null,
                    ],
                  ]
                }
              >
                test_body
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "flex": 1,
                "flexDirection": "row",
                "paddingBottom": 20,
              }
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "opacity": 1,
                  "paddingRight": 20,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#4C6070",
                      "fontSize": 23,
                    },
                    undefined,
                    Object {
                      "fontFamily": "anticon",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "fontFamily": "OpenSans-Regular",
                    },
                    null,
                  ]
                }
              >
                3
              </Text>
            </View>
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "opacity": 1,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#4C6070",
                      "fontSize": 21,
                    },
                    undefined,
                    Object {
                      "fontFamily": "anticon",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "fontFamily": "OpenSans-Regular",
                    },
                    null,
                  ]
                }
              >
                4
              </Text>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </RCTSafeAreaView>
</BVLinearGradient>
`;
